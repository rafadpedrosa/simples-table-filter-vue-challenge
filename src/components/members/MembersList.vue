<template>
  <b-table :fields="membersFields" class="mt-3 members-table" hover :items="membersList" show-empty caption-bottom empty-html="empty">
    <template #table-caption>
      <p v-if="membersList.length">Showing {{ membersList.length }} of {{ totalMembers }} members</p>
    </template>

    <template #cell(role)="{item: { role }}">
      <p class="editable-cell" >{{ role }}<b-icon class="pencil-icon mx-2" scale="1.2" icon="pencil" ></b-icon> </p>
    </template>

    <template #cell(projects)="{item: { projects }}">
      <p class="editable-cell" >{{ projects }}<b-icon class="pencil-icon mx-2" scale="1.2" icon="pencil" ></b-icon> </p>
    </template>

    <template #cell(payment)="{item: { payment, paymentDescription }}">
      <p class="editable-cell m-0" :class="{'emptyDescriptionValue' : !payment }">{{ paymentDescription }}<b-icon class="pencil-icon mx-2" scale="1.2"  icon="pencil" ></b-icon></p>
    </template>

    <template #cell(limits)="{item: { limits }}">
      <div class="d-flex editable-cell">
        <div>
          <p class="m-0" :class="{'emptyDescriptionValue' : !limits.weekly }">{{ limits.weeklyDescription }}</p>
          <p class="m-0" :class="{'emptyDescriptionValue' : !limits.daily }">{{ limits.dailyDescription }}</p>
        </div>
        <b-icon class="pencil-icon mx-2" scale="1.2"  icon="pencil" ></b-icon>
      </div>
    </template>

    <template #cell(time_tracking)="{item: { time_tracking }}">
      <h5><b-badge :variant="getTimeTrackingMessageType(time_tracking)">{{time_tracking}}</b-badge></h5>
    </template>

    <template #cell(actions) class="d-flex justify-content-end">
      <!-- actions are doing nothing -->
      <b-select class="table-select-actions"
                :options="[{ text: 'Actions', value: true },{ text: 'action 001', value: false }, { text: 'action 002', value: false }]"
                :value="true"></b-select>
    </template>

    <template #empty="scope">
      <h4>{{ scope.emptyText }} </h4>
    </template>
  </b-table>
</template>

<script>
export default {
  name: 'MembersList',
  props: {
    members: {
      type: Array
    },
    totalMembers: {
      type: Number
    }
  },
  methods: {
    getTimeTrackingMessageType (timeTracking) {
      return timeTracking === 'enabled' ? 'success' : 'danger'
    },
    resolveMembersDataAfterFetch (member) {
      const weeklyDescription = member.limits.weekly ? `${member.limits.weekly}:00 / Weekly` : 'No weekly limit'
      const dailyDescription = member.limits.daily ? `${member.limits.daily}:00 / Daily` : 'No daily limit'

      member.limits = { ...member.limits, weeklyDescription, dailyDescription }

      member.paymentDescription = member.payment || 'No pay rate / No bill rate'

      return member
    }
  },
  computed: {
    membersFields () {
      return [
        { key: 'name', label: 'Member' },
        'role',
        'projects',
        'payment',
        'limits',
        'time_tracking',
        { key: 'actions', label: '' }
      ]
    },
    membersList () {
      return (this.members || []).map(this.resolveMembersDataAfterFetch)
    }
  }
}
</script>

<style lang="scss">
.members-table > thead > tr > th {
    border-top: 0;
}

.members-table {
  .emptyDescriptionValue {
    color: $gray-600 default;
  }

  .table-select-actions {
    text-align: center;
    width: 10rem;
    min-width: 10rem;
    min-height: 20px;
    padding: 0 1.5rem 0 1rem;
    background-position-x: 83%;
    border-radius: 5px;
  }

  .editable-cell > .pencil-icon {
    display: none;
  }

  tr:hover {
    .editable-cell:hover {
      color: $primary;
      cursor: pointer;
    }
    .editable-cell {
      .pencil-icon {
        display: inline;
      }
    }
  }
}

</style>
