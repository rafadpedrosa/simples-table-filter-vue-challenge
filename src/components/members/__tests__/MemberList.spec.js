import { members, expected_member_modifications } from '@/../tests/dataMocks/members.mock.js'
import { mount } from '@vue/test-utils'
import MembersList from '@/components/members/MembersList'

const getMembersMock = () => members
/**
 * COMPONENT CONTRACT
 *
 * 1- It should render a table with a list of members
 * 1.1 - call the resolveMembersDataAfterFetch X times
 * 2- It should render a message if the table is empty
 * 2.1 - It should not call the resolveMembersDataAfterFetch
 * 3- It should have an resolvedObject
 **/

const spyOnResolveMembersDataAfterFetch = jest.spyOn(MembersList.methods, 'resolveMembersDataAfterFetch')

describe('MembersList.vue', () => {
  it('should NOT call the resolveMembersDataAfterFetch ', function () {
    const memberJared = getMembersMock()[0]

    const wrapper = mount(MembersList, { })

    const changedMemberJared = wrapper.vm.resolveMembersDataAfterFetch(memberJared)

    expect(changedMemberJared.limits.dailyDescription).toBe(expected_member_modifications.dailyDescription)
    expect(changedMemberJared.limits.weeklyDescription).toBe(expected_member_modifications.weeklyDescription)
    expect(changedMemberJared.paymentDescription).toBe(expected_member_modifications.paymentDescription)
  })

  it('should render a table with a list of members ', function () {
    spyOnResolveMembersDataAfterFetch.mockClear()

    const wrapper = mount(MembersList, { propsData: { members: getMembersMock() } })

    expect(spyOnResolveMembersDataAfterFetch).toHaveBeenCalledTimes(3)
    expect(wrapper).toMatchSnapshot()
  })

  it('should render a message if the table is empty ', function () {
    spyOnResolveMembersDataAfterFetch.mockReset()

    const wrapper = mount(MembersList, { propsData: { members: [] } })

    expect(spyOnResolveMembersDataAfterFetch).not.toBeCalled()
    expect(wrapper).toMatchSnapshot()
  })
})
