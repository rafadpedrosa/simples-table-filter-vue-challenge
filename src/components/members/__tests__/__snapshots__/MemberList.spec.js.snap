// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MembersList.vue should render a message if the table is empty  1`] = `
<table caption-bottom="" role="table" aria-busy="false" aria-colcount="7" class="table b-table mt-3 members-table">
  <caption></caption>
  <!---->
  <thead role="rowgroup" class="">
    <!---->
    <tr role="row" class="">
      <th role="columnheader" scope="col" aria-colindex="1" class="">
        <div>Member</div>
      </th>
      <th role="columnheader" scope="col" aria-colindex="2" class="">
        <div>Role</div>
      </th>
      <th role="columnheader" scope="col" aria-colindex="3" class="">
        <div>Projects</div>
      </th>
      <th role="columnheader" scope="col" aria-colindex="4" class="">
        <div>Payment</div>
      </th>
      <th role="columnheader" scope="col" aria-colindex="5" class="">
        <div>Limits</div>
      </th>
      <th role="columnheader" scope="col" aria-colindex="6" class="">
        <div>Time Tracking</div>
      </th>
      <th role="columnheader" scope="col" aria-colindex="7" aria-label="Actions" class="">
        <div></div>
      </th>
    </tr>
  </thead>
  <tbody role="rowgroup">
    <!---->
    <tr role="row" class="b-table-empty-row">
      <td colspan="7" role="cell" class="">
        <div role="alert" aria-live="polite">
          <h4>There are no records to show </h4>
        </div>
      </td>
    </tr>
    <!---->
  </tbody>
  <!---->
</table>
`;

exports[`MembersList.vue should render a table with a list of members  1`] = `
<table caption-bottom="" role="table" aria-busy="false" aria-colcount="7" class="table b-table mt-3 members-table table-hover">
  <caption>
    <p>Showing 3 of members</p>
  </caption>
  <!---->
  <thead role="rowgroup" class="">
    <!---->
    <tr role="row" class="">
      <th role="columnheader" scope="col" aria-colindex="1" class="">
        <div>Member</div>
      </th>
      <th role="columnheader" scope="col" aria-colindex="2" class="">
        <div>Role</div>
      </th>
      <th role="columnheader" scope="col" aria-colindex="3" class="">
        <div>Projects</div>
      </th>
      <th role="columnheader" scope="col" aria-colindex="4" class="">
        <div>Payment</div>
      </th>
      <th role="columnheader" scope="col" aria-colindex="5" class="">
        <div>Limits</div>
      </th>
      <th role="columnheader" scope="col" aria-colindex="6" class="">
        <div>Time Tracking</div>
      </th>
      <th role="columnheader" scope="col" aria-colindex="7" aria-label="Actions" class="">
        <div></div>
      </th>
    </tr>
  </thead>
  <tbody role="rowgroup">
    <!---->
    <tr role="row" class="">
      <td aria-colindex="1" role="cell" class="">Jared Brown</td>
      <td aria-colindex="2" role="cell" class="">
        <p class="editable-cell">Owner<svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil pencil-icon mx-2 b-icon bi">
            <g transform="translate(8 8) scale(1.2 1.2) translate(-8 -8)">
              <path fill-rule="evenodd" d="M11.293 1.293a1 1 0 0 1 1.414 0l2 2a1 1 0 0 1 0 1.414l-9 9a1 1 0 0 1-.39.242l-3 1a1 1 0 0 1-1.266-1.265l1-3a1 1 0 0 1 .242-.391l9-9zM12 2l2 2-9 9-3 1 1-3 9-9z"></path>
              <path fill-rule="evenodd" d="M12.146 6.354l-2.5-2.5.708-.708 2.5 2.5-.707.708zM3 10v.5a.5.5 0 0 0 .5.5H4v.5a.5.5 0 0 0 .5.5H5v.5a.5.5 0 0 0 .5.5H6v-1.5a.5.5 0 0 0-.5-.5H5v-.5a.5.5 0 0 0-.5-.5H3z"></path>
            </g>
          </svg></p>
      </td>
      <td aria-colindex="3" role="cell" class="">
        <p class="editable-cell">8<svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil pencil-icon mx-2 b-icon bi">
            <g transform="translate(8 8) scale(1.2 1.2) translate(-8 -8)">
              <path fill-rule="evenodd" d="M11.293 1.293a1 1 0 0 1 1.414 0l2 2a1 1 0 0 1 0 1.414l-9 9a1 1 0 0 1-.39.242l-3 1a1 1 0 0 1-1.266-1.265l1-3a1 1 0 0 1 .242-.391l9-9zM12 2l2 2-9 9-3 1 1-3 9-9z"></path>
              <path fill-rule="evenodd" d="M12.146 6.354l-2.5-2.5.708-.708 2.5 2.5-.707.708zM3 10v.5a.5.5 0 0 0 .5.5H4v.5a.5.5 0 0 0 .5.5H5v.5a.5.5 0 0 0 .5.5H6v-1.5a.5.5 0 0 0-.5-.5H5v-.5a.5.5 0 0 0-.5-.5H3z"></path>
            </g>
          </svg></p>
      </td>
      <td aria-colindex="4" role="cell" class="">
        <p class="editable-cell m-0 emptyDescriptionValue">No pay rate / No bill rate<svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil pencil-icon mx-2 b-icon bi">
            <g transform="translate(8 8) scale(1.2 1.2) translate(-8 -8)">
              <path fill-rule="evenodd" d="M11.293 1.293a1 1 0 0 1 1.414 0l2 2a1 1 0 0 1 0 1.414l-9 9a1 1 0 0 1-.39.242l-3 1a1 1 0 0 1-1.266-1.265l1-3a1 1 0 0 1 .242-.391l9-9zM12 2l2 2-9 9-3 1 1-3 9-9z"></path>
              <path fill-rule="evenodd" d="M12.146 6.354l-2.5-2.5.708-.708 2.5 2.5-.707.708zM3 10v.5a.5.5 0 0 0 .5.5H4v.5a.5.5 0 0 0 .5.5H5v.5a.5.5 0 0 0 .5.5H6v-1.5a.5.5 0 0 0-.5-.5H5v-.5a.5.5 0 0 0-.5-.5H3z"></path>
            </g>
          </svg></p>
      </td>
      <td aria-colindex="5" role="cell" class="">
        <div class="d-flex editable-cell">
          <div>
            <p class="m-0 emptyDescriptionValue">No weekly limit</p>
            <p class="m-0">8:00 / Daily</p>
          </div> <svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil pencil-icon mx-2 b-icon bi">
            <g transform="translate(8 8) scale(1.2 1.2) translate(-8 -8)">
              <path fill-rule="evenodd" d="M11.293 1.293a1 1 0 0 1 1.414 0l2 2a1 1 0 0 1 0 1.414l-9 9a1 1 0 0 1-.39.242l-3 1a1 1 0 0 1-1.266-1.265l1-3a1 1 0 0 1 .242-.391l9-9zM12 2l2 2-9 9-3 1 1-3 9-9z"></path>
              <path fill-rule="evenodd" d="M12.146 6.354l-2.5-2.5.708-.708 2.5 2.5-.707.708zM3 10v.5a.5.5 0 0 0 .5.5H4v.5a.5.5 0 0 0 .5.5H5v.5a.5.5 0 0 0 .5.5H6v-1.5a.5.5 0 0 0-.5-.5H5v-.5a.5.5 0 0 0-.5-.5H3z"></path>
            </g>
          </svg>
        </div>
      </td>
      <td aria-colindex="6" role="cell" class="">
        <h5><span class="badge badge-success">enabled</span></h5>
      </td>
      <td aria-colindex="7" role="cell" class=""><select class="table-select-actions custom-select">
          <option value="true">Actions</option>
          <option value="false">action 001</option>
          <option value="false">action 002</option>
        </select></td>
    </tr>
    <tr role="row" class="">
      <td aria-colindex="1" role="cell" class="">Adrian Goia</td>
      <td aria-colindex="2" role="cell" class="">
        <p class="editable-cell">Viewer<svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil pencil-icon mx-2 b-icon bi">
            <g transform="translate(8 8) scale(1.2 1.2) translate(-8 -8)">
              <path fill-rule="evenodd" d="M11.293 1.293a1 1 0 0 1 1.414 0l2 2a1 1 0 0 1 0 1.414l-9 9a1 1 0 0 1-.39.242l-3 1a1 1 0 0 1-1.266-1.265l1-3a1 1 0 0 1 .242-.391l9-9zM12 2l2 2-9 9-3 1 1-3 9-9z"></path>
              <path fill-rule="evenodd" d="M12.146 6.354l-2.5-2.5.708-.708 2.5 2.5-.707.708zM3 10v.5a.5.5 0 0 0 .5.5H4v.5a.5.5 0 0 0 .5.5H5v.5a.5.5 0 0 0 .5.5H6v-1.5a.5.5 0 0 0-.5-.5H5v-.5a.5.5 0 0 0-.5-.5H3z"></path>
            </g>
          </svg></p>
      </td>
      <td aria-colindex="3" role="cell" class="">
        <p class="editable-cell">5<svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil pencil-icon mx-2 b-icon bi">
            <g transform="translate(8 8) scale(1.2 1.2) translate(-8 -8)">
              <path fill-rule="evenodd" d="M11.293 1.293a1 1 0 0 1 1.414 0l2 2a1 1 0 0 1 0 1.414l-9 9a1 1 0 0 1-.39.242l-3 1a1 1 0 0 1-1.266-1.265l1-3a1 1 0 0 1 .242-.391l9-9zM12 2l2 2-9 9-3 1 1-3 9-9z"></path>
              <path fill-rule="evenodd" d="M12.146 6.354l-2.5-2.5.708-.708 2.5 2.5-.707.708zM3 10v.5a.5.5 0 0 0 .5.5H4v.5a.5.5 0 0 0 .5.5H5v.5a.5.5 0 0 0 .5.5H6v-1.5a.5.5 0 0 0-.5-.5H5v-.5a.5.5 0 0 0-.5-.5H3z"></path>
            </g>
          </svg></p>
      </td>
      <td aria-colindex="4" role="cell" class="">
        <p class="editable-cell m-0 emptyDescriptionValue">No pay rate / No bill rate<svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil pencil-icon mx-2 b-icon bi">
            <g transform="translate(8 8) scale(1.2 1.2) translate(-8 -8)">
              <path fill-rule="evenodd" d="M11.293 1.293a1 1 0 0 1 1.414 0l2 2a1 1 0 0 1 0 1.414l-9 9a1 1 0 0 1-.39.242l-3 1a1 1 0 0 1-1.266-1.265l1-3a1 1 0 0 1 .242-.391l9-9zM12 2l2 2-9 9-3 1 1-3 9-9z"></path>
              <path fill-rule="evenodd" d="M12.146 6.354l-2.5-2.5.708-.708 2.5 2.5-.707.708zM3 10v.5a.5.5 0 0 0 .5.5H4v.5a.5.5 0 0 0 .5.5H5v.5a.5.5 0 0 0 .5.5H6v-1.5a.5.5 0 0 0-.5-.5H5v-.5a.5.5 0 0 0-.5-.5H3z"></path>
            </g>
          </svg></p>
      </td>
      <td aria-colindex="5" role="cell" class="">
        <div class="d-flex editable-cell">
          <div>
            <p class="m-0">50:00 / Weekly</p>
            <p class="m-0">8:00 / Daily</p>
          </div> <svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil pencil-icon mx-2 b-icon bi">
            <g transform="translate(8 8) scale(1.2 1.2) translate(-8 -8)">
              <path fill-rule="evenodd" d="M11.293 1.293a1 1 0 0 1 1.414 0l2 2a1 1 0 0 1 0 1.414l-9 9a1 1 0 0 1-.39.242l-3 1a1 1 0 0 1-1.266-1.265l1-3a1 1 0 0 1 .242-.391l9-9zM12 2l2 2-9 9-3 1 1-3 9-9z"></path>
              <path fill-rule="evenodd" d="M12.146 6.354l-2.5-2.5.708-.708 2.5 2.5-.707.708zM3 10v.5a.5.5 0 0 0 .5.5H4v.5a.5.5 0 0 0 .5.5H5v.5a.5.5 0 0 0 .5.5H6v-1.5a.5.5 0 0 0-.5-.5H5v-.5a.5.5 0 0 0-.5-.5H3z"></path>
            </g>
          </svg>
        </div>
      </td>
      <td aria-colindex="6" role="cell" class="">
        <h5><span class="badge badge-success">enabled</span></h5>
      </td>
      <td aria-colindex="7" role="cell" class=""><select class="table-select-actions custom-select">
          <option value="true">Actions</option>
          <option value="false">action 001</option>
          <option value="false">action 002</option>
        </select></td>
    </tr>
    <tr role="row" class="">
      <td aria-colindex="1" role="cell" class="">Cody Rogers</td>
      <td aria-colindex="2" role="cell" class="">
        <p class="editable-cell">Viewer<svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil pencil-icon mx-2 b-icon bi">
            <g transform="translate(8 8) scale(1.2 1.2) translate(-8 -8)">
              <path fill-rule="evenodd" d="M11.293 1.293a1 1 0 0 1 1.414 0l2 2a1 1 0 0 1 0 1.414l-9 9a1 1 0 0 1-.39.242l-3 1a1 1 0 0 1-1.266-1.265l1-3a1 1 0 0 1 .242-.391l9-9zM12 2l2 2-9 9-3 1 1-3 9-9z"></path>
              <path fill-rule="evenodd" d="M12.146 6.354l-2.5-2.5.708-.708 2.5 2.5-.707.708zM3 10v.5a.5.5 0 0 0 .5.5H4v.5a.5.5 0 0 0 .5.5H5v.5a.5.5 0 0 0 .5.5H6v-1.5a.5.5 0 0 0-.5-.5H5v-.5a.5.5 0 0 0-.5-.5H3z"></path>
            </g>
          </svg></p>
      </td>
      <td aria-colindex="3" role="cell" class="">
        <p class="editable-cell">8<svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil pencil-icon mx-2 b-icon bi">
            <g transform="translate(8 8) scale(1.2 1.2) translate(-8 -8)">
              <path fill-rule="evenodd" d="M11.293 1.293a1 1 0 0 1 1.414 0l2 2a1 1 0 0 1 0 1.414l-9 9a1 1 0 0 1-.39.242l-3 1a1 1 0 0 1-1.266-1.265l1-3a1 1 0 0 1 .242-.391l9-9zM12 2l2 2-9 9-3 1 1-3 9-9z"></path>
              <path fill-rule="evenodd" d="M12.146 6.354l-2.5-2.5.708-.708 2.5 2.5-.707.708zM3 10v.5a.5.5 0 0 0 .5.5H4v.5a.5.5 0 0 0 .5.5H5v.5a.5.5 0 0 0 .5.5H6v-1.5a.5.5 0 0 0-.5-.5H5v-.5a.5.5 0 0 0-.5-.5H3z"></path>
            </g>
          </svg></p>
      </td>
      <td aria-colindex="4" role="cell" class="">
        <p class="editable-cell m-0 emptyDescriptionValue">No pay rate / No bill rate<svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil pencil-icon mx-2 b-icon bi">
            <g transform="translate(8 8) scale(1.2 1.2) translate(-8 -8)">
              <path fill-rule="evenodd" d="M11.293 1.293a1 1 0 0 1 1.414 0l2 2a1 1 0 0 1 0 1.414l-9 9a1 1 0 0 1-.39.242l-3 1a1 1 0 0 1-1.266-1.265l1-3a1 1 0 0 1 .242-.391l9-9zM12 2l2 2-9 9-3 1 1-3 9-9z"></path>
              <path fill-rule="evenodd" d="M12.146 6.354l-2.5-2.5.708-.708 2.5 2.5-.707.708zM3 10v.5a.5.5 0 0 0 .5.5H4v.5a.5.5 0 0 0 .5.5H5v.5a.5.5 0 0 0 .5.5H6v-1.5a.5.5 0 0 0-.5-.5H5v-.5a.5.5 0 0 0-.5-.5H3z"></path>
            </g>
          </svg></p>
      </td>
      <td aria-colindex="5" role="cell" class="">
        <div class="d-flex editable-cell">
          <div>
            <p class="m-0">40:00 / Weekly</p>
            <p class="m-0 emptyDescriptionValue">No daily limit</p>
          </div> <svg viewBox="0 0 16 16" width="1em" height="1em" focusable="false" role="img" aria-label="pencil" xmlns="http://www.w3.org/2000/svg" fill="currentColor" class="bi-pencil pencil-icon mx-2 b-icon bi">
            <g transform="translate(8 8) scale(1.2 1.2) translate(-8 -8)">
              <path fill-rule="evenodd" d="M11.293 1.293a1 1 0 0 1 1.414 0l2 2a1 1 0 0 1 0 1.414l-9 9a1 1 0 0 1-.39.242l-3 1a1 1 0 0 1-1.266-1.265l1-3a1 1 0 0 1 .242-.391l9-9zM12 2l2 2-9 9-3 1 1-3 9-9z"></path>
              <path fill-rule="evenodd" d="M12.146 6.354l-2.5-2.5.708-.708 2.5 2.5-.707.708zM3 10v.5a.5.5 0 0 0 .5.5H4v.5a.5.5 0 0 0 .5.5H5v.5a.5.5 0 0 0 .5.5H6v-1.5a.5.5 0 0 0-.5-.5H5v-.5a.5.5 0 0 0-.5-.5H3z"></path>
            </g>
          </svg>
        </div>
      </td>
      <td aria-colindex="6" role="cell" class="">
        <h5><span class="badge badge-success">enabled</span></h5>
      </td>
      <td aria-colindex="7" role="cell" class=""><select class="table-select-actions custom-select">
          <option value="true">Actions</option>
          <option value="false">action 001</option>
          <option value="false">action 002</option>
        </select></td>
    </tr>
    <!---->
    <!---->
  </tbody>
  <!---->
</table>
`;
